specs/validator/0_beacon-chain-validator.md

```diff
@@ -339,6 +350,81 @@ def get_signed_attestation_data(state: BeaconState, attestation: IndexedAttestat
     return bls_sign(privkey, hash_tree_root(attestation_data_and_custody_bit), domain)
 ```
 
+#### Broadcast attestation
+
+Finally, the validator broadcasts `attestation` to the associated attestation subnet -- the `index{attestation.data.index % ATTESTATION_SUBNET_COUNT}_beacon_attestation` pubsub topic.
+
+### Attestation aggregation
+
+Some validators are selected to locally aggregate attestations with a similar `attestation_data` to their constructed `attestation` for the assigned `slot`.
+
+#### Aggregation selection
+
+A validator is selected to aggregate based upon the return value of `is_aggregator()`.
+
+```python
+def slot_signature(state: BeaconState, slot: Slot, privkey: int) -> BLSSignature:
+    domain = get_domain(state, DOMAIN_BEACON_ATTESTER, compute_epoch_at_slot(slot))
+    return bls_sign(privkey, hash_tree_root(slot), domain)
+```
+
+```python
+def is_aggregator(state: BeaconState, slot: Slot, index: CommitteeIndex, slot_signature: BLSSignature) -> bool:
+    committee = get_beacon_committee(state, slot, index)
+    modulo = max(1, len(committee) // TARGET_AGGREGATORS_PER_COMMITTEE)
+    return bytes_to_int(hash(slot_signature)[0:8]) % modulo == 0
```

Comment:
Is it possible that there's *no any aggregator* gets selected from the committee even the committee is not empty?
How about just naively using the "index of validator in committee" to select the aggregators? 

```python
def is_aggregator(state: BeaconState, slot: Slot, index: CommitteeIndex, slot_signature: BLSSignature, validator_index: ValidatorIndex) -> bool:
    committee = get_beacon_committee(state, slot, index)
    modulo = max(1, len(committee) // TARGET_AGGREGATORS_PER_COMMITTEE)
    validator_index_in_committee = committee.index(validator_index)
    return validator_index_in_committee % module == 0
```

Reply:
FWIW, I made a chart: https://docs.google.com/spreadsheets/d/1C7pBqEWJgzk3_jesLkqJoDTnjZOODnGTOJUrxUMdxMA/edit?usp=sharing

- If `TARGET_AGGREGATORS_PER_COMMITTEE = 16`, the probability is close to `1.06E-07` for the worst case.
- If `TARGET_AGGREGATORS_PER_COMMITTEE = 32`, the probability is close to `9.84E-15` for the worst case.

The probability looks fine, at least for now. We can adjust `TARGET_AGGREGATORS_PER_COMMITTEE` when committee size getting larger anyway.